routes/degree_days.py

```diff
@@ -0,0 +1,375 @@
+            )
+        for tempstat in range(0, 3):
+            request_str = get_dd_wcps_request_str(
+                x, y, cov_id, "1:2", timestring, tempstat
+            )
+            point_data_list.append(
+                await fetch_data([generate_wcs_query_url(request_str)])
+            )
+
+    return point_data_list
+
+
+def create_csv(data_pkg, var_ep, place_id=None, lat=None, lon=None):
+    """Create CSV file with metadata string and location based filename.
+    Args:
+        data_pkg (dict): JSON-like object of data
+        var_ep (str): heating, below_zero, thawing_index, or freezing_index
+        place_id: place identifier (e.g., AK124)
+        lat: latitude for points or None for polygons
+        lon: longitude for points or None for polygons
+    Returns:
+        CSV response object
+    """
+
+    fieldnames = [
+        "model",
+        "year",
+        "variable",
+        "value",
+    ]
+
+    csv_dicts = build_csv_dicts(
+        data_pkg,
+        fieldnames,
+    )
+
+    if var_ep == "heating":
+        metadata = (
+            "# dd is the total annual degree days below 65°F for the specified model\n"
+        )
+    elif var_ep == "below_zero":
+        metadata = (
+            "# dd is the total annual degree days below 0°F for the specified model\n"
+        )
+    elif var_ep == "thawing_index":
+        metadata = "# dd is the total annual degree days above freezing for the specified model\n"
+    elif var_ep == "freezing_index":
+        metadata = "# dd is the total annual degree days below freezing for the specified model\n"
+
+    filename = var_label_lu[var_ep] + " for " + lat + ", " + lon + ".csv"
+
+    return write_csv(csv_dicts, fieldnames, filename, metadata)
+
+
+@routes.route("/mmm/degree_days/<var_ep>/<horp>/<lat>/<lon>")
+@routes.route("/mmm/degree_days/<var_ep>/<horp>/<lat>/<lon>/<start_year>/<end_year>")
+def run_fetch_dd_point_data(var_ep, lat, lon, horp, start_year=None, end_year=None):
+    """Point data endpoint. Fetch point data for
+    specified lat/lon and return JSON-like dict.
+
+    Args:
+        var_ep (str): heating, below_zero, thawing_index, or freezing_index
+        lat (float): latitude
+        lon (float): longitude
+
+    Returns:
+        JSON-like dict of requested degree days data
+
+    Notes:
+        example request: http://localhost:5000/degree_days/heating/all/65/-147
+    """
+    validation = validate_latlon(lat, lon)
+    if validation == 400:
+        return render_template("400/bad_request.html"), 400
+    if validation == 422:
+        return (
+            render_template(
+                "422/invalid_latlon.html", west_bbox=WEST_BBOX, east_bbox=EAST_BBOX
+            ),
+            422,
+        )
+
+    x, y = project_latlon(lat, lon, 3338)
+
+    if start_year is not None:
+        if end_year is not None:
+            validation = validate_year(start_year, end_year)
+
+            if validation == 400:
```

Comment:
Wondering if maybe we want to use the new `422/invalid_year.html` here instead of the vanilla bad request. There are some "invalid year" flavor requests that get captured here before making it down to line 346, e.g. http://localhost:5000/mmm/degree_days/heating/hp/65.0628/-146.1627/1850/2000

Reply:
Fixed in 54c15f65bb9ac357bcd0bfef06927e2f3ea39c3e.